src/main/java/uk/gov/hmcts/reform/prl/services/StmtOfServImplService.java

```diff
@@ -342,4 +356,156 @@ private String getPartyIds(String caseTypeOfApplication,
                                .map(Element::getValue)
                                .toList());
     }
+
+    public void saveCitizenSos(String caseId, String eventId,String authorisation, CitizenSos sosObject) {
+        StartAllTabsUpdateDataContent startAllTabsUpdateDataContent
+            = allTabService.getStartUpdateForSpecificEvent(caseId, eventId);
+        Map<String, Object> updatedCaseDataMap = startAllTabsUpdateDataContent.caseDataMap();
+        CaseData updatedCaseData = startAllTabsUpdateDataContent.caseData();
+        log.info("Unserved respondent pack {}", updatedCaseDataMap.get(UN_SERVED_RESPONDENT_PACK));
+        if (null != updatedCaseDataMap.get(UN_SERVED_RESPONDENT_PACK)
+            && CollectionUtils.isNotEmpty(updatedCaseData.getServiceOfApplication().getUnServedRespondentPack().getPackDocument())) {
+            List<Element<StmtOfServiceAddRecipient>> stmtOfServiceforApplication = new ArrayList<>();
+            updateStatementOfServiceCollection(sosObject, updatedCaseData, stmtOfServiceforApplication);
+            updatedCaseDataMap.put(STMT_OF_SERVICE_FOR_APPLICATION, stmtOfServiceforApplication);
+            if (YesOrNo.Yes.getDisplayedValue().equalsIgnoreCase(sosObject.getIsOrder())) {
+                List<Element<ServedApplicationDetails>> finalServedApplicationDetailsList;
+                List<Element<BulkPrintDetails>> bulkPrintDetails = new ArrayList<>();
+                finalServedApplicationDetailsList = updateFinalListOfServedApplications(
```

Comment:
AccessCode Notifications sent to respondent should be recorded as part of a collection and stored in db

Reply:
Also check contact preference before sending access code letter